### PR TITLE
add vmware-exporter chart

### PR DESCRIPTION
**Pull request by mihiragrawal**
**What this PR does / why we need it**: To add basic monitoring of vsphere provider. This PR adds a chart named `vmware-exporter`, which adds metrics for Prometheus. According to the upstream chart below information will be get. 

- Basic VM and Host metrics
- Current number of active snapshots
- Datastore size and other stuff
- Snapshot Unix timestamp creation date

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**: 

**Documentation**:

**Does this PR introduce a user-facing change?**:

```
NONE
```

### CODE REVIEW

File: charts/monitoring/vmware-exporter/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "vmware-exporter.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
```
Comment (wurbanski) [inline charts/monitoring/vmware-exporter/templates/_helpers.tpl]: **Note: this applies to any instance where you use `.Values.something`**

Since you updated the values.yaml to be prefixed with `vmware-exporter`, all instances of `.Values.xxx` in templates have to be changed to `.Values.vmware-exporter.xxx`

File: charts/monitoring/vmware-exporter/values.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    #  readOnlyRootFilesystem: true
+    #  runAsGroup: 65532
+    #  runAsNonRoot: true
+    #  runAsUser: 65532
+  
+  # Set the Pod security context
+  podSecurityContext:
+    {}
+    #  fsGroup: 65532
+
```
Comment (wurbanski) [inline charts/monitoring/vmware-exporter/values.yaml]: nitpicking, but since there are some changes to do already: can you fix this newline situation here? :)

**Review by wurbanski — CHANGES_REQUESTED** [review]
@mihiragrawal Can you describe the scope of changes you made to the original chart?

File: charts/monitoring/vmware-exporter/templates/_helpers.tpl (hunk 1 above)
Comment (mihiragrawal) [reply charts/monitoring/vmware-exporter/templates/_helpers.tpl]: Thanks for pointing it out. Fixed.

File: charts/monitoring/vmware-exporter/values.yaml (hunk 2 above)
Comment (mihiragrawal) [reply charts/monitoring/vmware-exporter/values.yaml]: Fixed.

**Comment by xrstf** [comment]
> - Inject license information.

Would not be needed if we just used the upstream chart directly. 

> - add parent value string `vmware-exporter` in `value.yaml` to adjust values in KKP value file.

That is nice.

> - Update image tag to latest docker image available on [dockerhub](https://hub.docker.com/r/pryorda/vmware_exporter/tags).

We could simply contribute a one-line change to the upstream chart :)

> - Enabled svc deployment.
> - Update replica count to 2.

That can just be done via the values.yaml, this is not really a "change".

> - cleanup non required files, like `createRelease.sh` and `owner`.....

Would not hurt us if we just used the upstream chart.

I don't think we should fork the chart, as we're on our way to get rid of most of our forks and use upstream instead. I would really not want to introduce new charts that are immediately deprecated. Nothing of this is added to the KKP installer, so this chart would have to be installed manually anyway.

For these reasons I would strongly vote towards just recommending to use the upstream chart. It does bring a bit of inconvenience, because the upstream chart is not ready to be used with our old "single values.yaml for everything"-technique, but that technique is also on its way out as we begin to use more upstream charts.

**Comment by toschneck** [comment]
> For these reasons I would strongly vote towards just recommending to use the upstream chart. It does bring a bit of inconvenience, because the upstream chart is not ready to be used with our old "single values.yaml for everything"-technique, but that technique is also on its way out as we begin to use more upstream charts.

That's pretty new to  me. So do we have a concept how we manage the dependency of a upstream chart? 
For me it just should be easy to use a "predefined" best practices stack, what should be versioned. Versioned tied charts are stable and help us to bring up customer setups quickly. 

@xrstf  how you will ensure that we have a standard set of upstream components (charts / operators etc.) easy manageable for our customer?

**Comment by xrstf** [comment]
> ow you will ensure that we have a standard set of upstream components (charts / operators etc.) easy manageable for our customer?

That can be as easy as a `map[string]string` in the KKP installer, listing the charts and their version. Or a config file somewhere.

Also, I would not want us to extend the set of charts we support/recommend. We are not in the business of being a Helm chart broker and shipping tons of charts; every new chart is another dependency we need to carefully consider.

See also https://github.com/kubermatic/kubermatic/issues/7897 and our #dev-hackathon-helm-squad Slack channel for more info. I am not in the team that manages the transition from custom charts to upstream charts ;)

**Comment by shaase-ctrl** [comment]
How can we support that there are no configs we "forget"?

**Comment by xrstf** [comment]
Can you rephrase that question?
